test/parse.test.js

```diff
@@ -46,6 +46,40 @@ describe('Parse', () => {
     expect(dayjs(time).valueOf()).toBe(moment(time).valueOf())
   })
 
+  it('String RFC 2822, time and zone', () => {
+    const time = 'Mon, 11 Feb 2019 09:46:50 GMT+1'
+    const expected = '2019-02-11T08:46:50.000Z'
+    const d = dayjs(time)
+    expect(d.toISOString()).toEqual(expected)
+    expect(d.valueOf()).toBe(moment(time).valueOf())
+  })
+
+  it('String ECMAScript, time and zone', () => {
+    // should parse dates formatted in ECMA script format
+    // see https://www.ecma-international.org/ecma-262/9.0/index.html#sec-date.prototype.tostring
+    const time = 'Mon Feb 11 2019 11:01:37 GMT+0100 (Mitteleuropäische Normalzeit)'
+    const expected = '2019-02-11T10:01:37.000Z'
+    const d = dayjs(time)
+    expect(d.toISOString()).toEqual(expected)
+    expect(d.valueOf()).toBe(moment(time).valueOf())
+  })
+
+  it('rejects invalid values', () => {
+    expect(dayjs({}).isValid()).toBe(false)
+    expect(dayjs(() => '2018-01-01').isValid()).toBe(false)
+    expect(dayjs(Infinity).isValid()).toBe(false)
+    expect(dayjs(NaN).isValid()).toBe(false)
+    expect(dayjs([2018, 5, 1, 13, 52, 44]).isValid()).toBe(false) // Arrays with time part
+  })
+
+  it('parses Arrays with date part', () => {
+    const dateParts = [2018, 5, 1]
+    const expected = '2018-05-01T00:00:00.000Z'
+    const d = dayjs(dateParts)
+    const normalized = d.add(d.utcOffset(), 'minutes') // make test run in every timezone
```

Comment:
what's this test for?

Reply:
Making implicit behavior explicit: Arrays fall through to the default `Date` constructor and should work according to ECMAScript standard (but only for the date part, time is not supported), so I wanted to assert that here explicitly. Testing this is a first step to also documenting that behavior or to change the behavior and be more strict than standard Javascript is. `moment.js` is way more opinionated about these cases than `dayjs` is, but it is hard to spot the differences.

Reply:
Moment.js will handle that case differently than dayjs will:

```
const arr = [2018, 5, 1];
const dayjsDate = moment([2018, 5, 1]);
const momentDate = dayjs([2018, 5, 1]);
console.log(`dayjs: ${dayjsDate.toISOString()}`)
console.log(`momentjs: ${momentDate.toISOString()}`)
```

yields

```
dayjs: 2018-05-31T22:00:00.000Z
momentjs: 2018-04-30T22:00:00.000Z
```

So I wanted to assert that explicitly here.

Reply:
actually I'm asking why ` make test run in every timezone` 😬

Reply:
If you parse a date without tz information (which can not be avoided with the Array), it will parse it as if it were your local timezone. So depending on in which timezone you run the test, you would get different results. To avoid that, I normalize the result by adding the UTC offset so I can expect against a fixed GMT timezone string.

Reply:
Got you, but a little bit hard to understand.

Can we just compare `format()` string or unix timestamp instead?

Reply:
But it's a whole different date, isn't it? I am adding seconds to make it run, after all... not just reformatting it? 

If I take this example and run it locally, it actually assumes GMT+2 (because 2018-05-01 in Germany is summertime):

```
new Date([2018, 5, 1]).toUTCString()
"Mon, 30 Apr 2018 22:00:00 GMT"
```
So depending on where you run this code, it will assume the GMT time to be relative to the timezone that will be valid in that place at that point in time. Just correcting the timezone via its offset seems to be the easiest way? But maybe I am missing something?

Reply:
oh I see. You are right.

Reply:
I struggled with this a lot during work on a project that made heavy use of dates (my task was to switch from momentjs to dayjs writing a million tests for everything). Those tests were all fine on my machine but failed on machines of colleagues in different timezones, so I wrote an abstraction that kind of does what my test here is doing (adjusting the timezone by offset). It is really not easy getting all of that right :/.

Reply:
@neemzy seems the test we made
```
expect dayjs() tobe moment()
```
just skipped this issue 😬